presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardOrganizationManager.java

```diff
@@ -0,0 +1,197 @@
+{
+    private IDBI dbi;
+    private Handle dummyHandle;
+    private MetadataDao metadataDao;
+    private ShardOrganizerDao organizerDao;
+
+    private static final Table tableInfo = new Table(1L, OptionalLong.empty(), OptionalInt.empty(), OptionalLong.empty());
+    private static final Table temporalTableInfo = new Table(1L, OptionalLong.empty(), OptionalInt.empty(), OptionalLong.of(1));
+
+    private static final List<Type> types = ImmutableList.of(BIGINT, VARCHAR, DATE, TIMESTAMP);
+
+    @BeforeMethod
+    public void setup()
+    {
+        dbi = new DBI("jdbc:h2:mem:test" + System.nanoTime());
+        dummyHandle = dbi.open();
+        metadataDao = dbi.onDemand(MetadataDao.class);
+        organizerDao = dbi.onDemand(ShardOrganizerDao.class);
+
+        createTablesWithRetry(dbi);
+    }
+
+    @AfterMethod
+    public void teardown()
+    {
+        dummyHandle.close();
+    }
+
+    @Test
+    public void testOrganizationEligibleTables()
+            throws Exception
+    {
+        long table1 = metadataDao.insertTable("schema", "table1", false, true, null);
+        metadataDao.insertColumn(table1, 1, "foo", 1, "bigint", 1, null);
+
+        metadataDao.insertTable("schema", "table2", false, true, null);
+        metadataDao.insertTable("schema", "table3", false, false, null);
+        assertEquals(metadataDao.getOrganizationEligibleTables(), ImmutableSet.of(table1));
+    }
+
+    @Test(timeOut = 1_000)
+    public void testTableDiscovery()
+            throws Exception
+    {
+        long table1 = metadataDao.insertTable("schema", "table1", false, true, null);
+        metadataDao.insertColumn(table1, 1, "foo", 1, "bigint", 1, null);
+
+        long table2 = metadataDao.insertTable("schema", "table2", false, true, null);
+        metadataDao.insertColumn(table2, 1, "foo", 1, "bigint", 1, null);
+
+        metadataDao.insertTable("schema", "table3", false, false, null);
+
+        ShardOrganizationManager organizationManager = createShardOrganizationManager("100ms");
+
+        // initializes tables
+        Set<Long> actual = organizationManager.discoverAndInitializeTablesToOrganize();
+        assertEquals(actual, ImmutableSet.of(table1, table2));
+
+        // update the start times and test that the tables are discovered after interval seconds
+        organizerDao.updateLastStartTime("node1", table1);
+        organizerDao.updateLastStartTime("node1", table2);
+
+        // wait for some time (interval time) for the tables to be eligible for organization
+        while (organizationManager.discoverAndInitializeTablesToOrganize().isEmpty()) {
+            MILLISECONDS.sleep(10);
+        }
+        assertEquals(organizationManager.discoverAndInitializeTablesToOrganize(), ImmutableSet.of(table1, table2));
+    }
+
+    @Test
+    public void testSimple()
+            throws Exception
+    {
+        long timestamp = Duration.ofDays(Duration.ofNanos(System.nanoTime()).toDays()).toMillis();
```

Comment:
`Duration.ofNanos(System.nanoTime())` is weird since the initial value of `System.nanoTime()` is effectively a random number. If you want a random number, use that explicitly. However, it's better for a test like this to be deterministic using a fixed value.

(A more typical use for random numbers would be a fuzz test generating a million random items.)